crates/ursa-rpc-service/src/api.rs

```diff
@@ -79,68 +88,228 @@ pub trait NetworkInterface: Sync + Send + 'static {
     /// Put a car file and start providing to the network
     async fn put_car<R: AsyncRead + Send + Unpin>(&self, file: Car<R>) -> Result<Vec<Cid>>;
 
-    // Put a file using a local path
+    /// Put a file using a local path
     async fn put_file(&self, path: String) -> Result<Vec<Cid>>;
 
-    // get peers from the network
+    /// Get peers from the network
     async fn get_peers(&self) -> Result<HashSet<PeerId>>;
 
-    // get the addrresses that p2p node is listening on
+    /// Get the addresses that p2p node is listening on
     async fn get_listener_addresses(&self) -> Result<Vec<Multiaddr>>;
 }
-#[derive(Clone)]
+
+type PendingRequests = Arc<RwLock<HashMap<Cid, Vec<Sender<Result<()>>>>>>;
+
 pub struct NodeNetworkInterface<S>
 where
     S: Blockstore + Store + Send + Sync + 'static,
 {
     pub store: Arc<UrsaStore<S>>,
     pub network_send: Sender<NetworkCommand>,
     pub provider_send: Sender<ProviderCommand>,
+    pending_requests: PendingRequests,
+    client: Arc<Client>,
+    origin_config: OriginConfig,
 }
 
-#[async_trait]
-impl<S> NetworkInterface for NodeNetworkInterface<S>
+impl<S> NodeNetworkInterface<S>
```

Comment:
Why was the async trait removed and `NetworkInterface` impl as well

Reply:
it's not it's just the diff being weird. It's the internal implementation added above the trait implementation

Reply:
Like did you move the block around is that why the diff is being weird?

Reply:
no nothing was moved just inserted in between

Reply:
for better readability which maintains structure when the structure is not required to change. makes reviewing hard

Reply:
I can move it for you

Reply:
I think in general not just for me viewing pleasure. For everyone's

Reply:
https://github.com/fleek-network/ursa/pull/278/commits/189110264aa4e4398f5b0b5eafa5dec0cb61a20e